parity-clib/python/parity/__init__.py

```diff
@@ -0,0 +1,121 @@
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Parity.  If not, see <http://www.gnu.org/licenses/>.
+
+from queue import Queue
+
+# Rust extension module, see src/python.rs
+import _parity
+
+
+class _CallbackGenerator(object):
+    """Thin wrapper around a Queue which can be passed as a callback to _parity.* functions"""
+
+    def __init__(self):
+        self.queue = Queue(maxsize=1)
+
+    def __call__(self, value):
+        self.queue.put(value)
+
+    def __iter__(self):
+        return self
+
+    def __next__(self):
+        return self.get()
+
+    def get(self, block=True, timeout=None):
+        """Get an element from the queue
+
+        :param block: Should we block if no element available
+        :param timeout: Time to wait for new element
+        :return: Top item from the queue
+        :raises: queue.Empty
+        """
+        return self.queue.get(block, timeout)
+
+    def get_nowait(self):
+        """Get an element from the queue, do not wait.
+
+        Equivalent to `get(False)`
+
+        :return: Top item from the queue
+        :raises: queue.Empty
+        """
+        return self.queue.get_nowait()
+
+
+class Subscription(object):
+    """Encapsulates a subscription returned from subscribe_ws, allowing iteration over events and unsubscribing"""
+
+    def __init__(self, sub, events):
+        self._sub = sub
+        self.events = events
+
+    def unsubscribe(self):
+        """Unsubscribe from the underlying subscription"""
+        self._sub.unsubscribe()
+
+
+class Parity(object):
+    """Connection to Parity client"""
+
+    def __init__(self, options, logger_mode='', logger_file=''):
+        """Configure and start Parity
+
+        :param options: Command line arguments to pass to Parity
+        :param logger_mode: Logger options to pass to Parity
+        :param logger_file: File to log to
+        """
+        config = _parity.config_from_cli(options)
+        self.handle = _parity.build(config, logger_mode, logger_file)
+
+    def rpc_query_cb(self, query, cb, timeout_ms=1000):
+        """Perform a RPC query, return immediately, cb will be invoked with the result asynchronously
+
+        :param query: Query to perform
+        :param cb: Callback to invoke with results
+        :param timeout_ms: Timeout in milliseconds
+        """
+        _parity.rpc_query(self.handle, query, timeout_ms, cb)
+
+    def rpc_query(self, query, timeout_ms=1000):
+        """Perform a RPC query and return the result synchronously
+
+        :param query: Query to perform
+        :param timeout_ms: Timeout in milliseconds
+        :return: Result of the rpc call
+        """
+        cb = _CallbackGenerator()
+        self.rpc_query_cb(query, cb, timeout_ms)
+        return next(cb)
```

Comment:
This will call the iterator of the Callback class?

Reply:
of the `_CallbackGenerator` class, yes